pkg/apis/hive/v1alpha1/syncset_types.go

```diff
@@ -65,6 +65,40 @@ type SyncObjectPatch struct {
 	PatchType string `json:"patchType,omitempty"`
 }
 
+// SyncResource represents an existing object to be synced
+type SyncResource struct {
+	Source SyncResourceSource `json:"source"`
+	Target SyncResourceTarget `json:"target"`
+}
+
+// SyncResourceSource describes a source object to sync to clusters SyncSet applies to
+type SyncResourceSource struct {
```

Comment:
Why not use `ObjectReference` instead of defining new types? That is the general guidance from the API conventions. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#references-to-related-objects

Reply:
+1 for ObjectReference, although I actually envisioned that this would be specifically for Secrets. We already built a mechanism for generic objects, it's just a matter of Secret data. Do we think there's a strong driver to also allow the copy of generic resources? We'd then have two ways to do that.

Reply:
I've changed the `source` to be an `ObjectReference` but we still want to keep the idea of a target in the event that namespace/name will not be the same as they are currently, correct?

Reply:
We want to keep the `target` field, but it should be an `ObjectReference` too.

Reply:
Okay 👍